### ISSUE TITLE
Introduce shift swap request notification follow-ups

### ISSUE THREAD

**Issue by joeyorlando** [grafana/oncall#2679]
### What would you like to see!

This is a follow-up task to #2630 and #2588. **Note**: both of those issues should be merged first before working on this.

Those tasks were focused on sending out the initial notifications when a shift swap request is first created.

**Deliverables**
In this task, we should implement a strategy to send intelligent follow-up Slack, and mobile app, notifications.

First we need to come to a consensus on a notification follow-up schedule (ex. First notification <= 1 week before the first shift starts. Second X time before shift starts, etc).

### Product Area

Chatops, Mobile App, Schedules

**Comment by joeyorlando** [comment]
@vadimkerr and I decided to first merge #2630 and #2588 before working on follow-ups because Slack and the mobile-app will have a lot of shared logic for calculating when to send these notifications
